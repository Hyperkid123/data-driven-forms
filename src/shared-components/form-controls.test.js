import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Button as Pf3Button } from 'patternfly-react';
import { Button as Pf4Button } from '@patternfly/react-core';
import FormControls from './form-controls';
import ComponentType, { configureContext } from '../renderer-context';

describe('<FormControls />', () => {
  let initialProps;
  beforeEach(() => {
    initialProps = {
      onSubmit: jest.fn(),
      onReset: jest.fn(),
      onCancel: jest.fn(),
    };
  });
  it('should render all pf3 form controls', () => {
    const wrapper = mount(
      <ComponentType.Provider value={ configureContext({ formType: 'pf3' }) }>
        <FormControls { ...initialProps } />
      </ComponentType.Provider>
    );
    const buttons = wrapper.find(Pf3Button);
    expect(buttons).toHaveLength(3);
  });

  it('should render all form controls with custom labels', () => {
    const labels = {
      submitLabel: 'Foo',
      cancelLabel: 'Bar',
      resetLabel: 'quax',
    };
    const wrapper = mount(
      <ComponentType.Provider value={ configureContext({ formType: 'pf3' }) }>
        <FormControls { ...initialProps } { ...labels } />
      </ComponentType.Provider>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render all pf4 form controls', () => {
    const wrapper = mount(
      <ComponentType.Provider value={ configureContext({ formType: 'pf4' }) }>
        <FormControls { ...initialProps } />
      </ComponentType.Provider>
    );
    const buttons = wrapper.find(Pf4Button);
    expect(buttons).toHaveLength(3);
  });
});
