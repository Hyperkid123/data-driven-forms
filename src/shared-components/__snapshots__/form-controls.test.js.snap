// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormControls /> should render all form controls with custom labels 1`] = `
<FormControls
  canReset={false}
  canSubmit={false}
  cancelLabel="Bar"
  onCancel={[MockFunction]}
  onReset={[MockFunction]}
  onSubmit={[MockFunction]}
  resetLabel="quax"
  submitLabel="Foo"
>
  <Col
    bsClass="col"
    componentClass="div"
    xs={12}
  >
    <div
      className="col-xs-12"
    >
      <FormGroup
        bsClass="form-group"
      >
        <div
          className="form-group"
        >
          <ButtonGroup
            block={false}
            bsClass="pull-right"
            justified={false}
            vertical={false}
          >
            <div
              className="pull-right"
            >
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsStyle="primary"
                disabled={true}
                onClick={[MockFunction]}
                type="button"
              >
                <button
                  className="btn btn-primary"
                  disabled={true}
                  onClick={[MockFunction]}
                  type="button"
                >
                  Foo
                </button>
              </Button>
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsStyle="default"
                disabled={false}
                onClick={[MockFunction]}
                type="button"
              >
                <button
                  className="btn btn-default"
                  disabled={false}
                  onClick={[MockFunction]}
                  type="button"
                >
                  Bar
                </button>
              </Button>
            </div>
          </ButtonGroup>
        </div>
      </FormGroup>
    </div>
  </Col>
</FormControls>
`;
